plugin/storage/es/factory.go

```diff
@@ -83,6 +83,39 @@ func NewFactory() *Factory {
 	}
 }
 
+func NewFactoryWithConfig(
+	cfg config.Configuration,
+	metricsFactory metrics.Factory,
+	logger *zap.Logger,
+) (*Factory, error) {
+	if err := cfg.Validate(); err != nil {
+		return nil, err
+	}
+
+	cfg.MaxDocCount = defaultMaxDocCount
+	cfg.Enabled = true
+
+	archive := make(map[string]*namespaceConfig)
+	archive[archiveNamespace] = &namespaceConfig{
+		Configuration: cfg,
+		namespace:     archiveNamespace,
+	}
```

Comment:
ok, this is logically wrong, but will work for now. The underlying problem is the existence of `storage.ArchiveFactory` interface, we need to get rid of it long term, because as far as interface goes the archive storage is just like a regular storage, only with different configuration.

Reply:
#5230